source/include/dqm4hep/MonitorElement.h

```diff
@@ -412,10 +420,21 @@ namespace dqm4hep {
     //-------------------------------------------------------------------------------------------------
 
     template <typename T>
-    void TScalarObject<T>::Init() {
+    inline void TScalarObject<T>::Init() {
       m_scalar = 0;
       m_pPaveText = nullptr;
     }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    template <typename T>
+    inline Bool_t TScalarObject<T>::IsEqual(const TObject *other) const {
+      const auto *otherScalar = (TScalarObject<T>*)other;
```

Comment:
You can remove the c style cast if you define object as TScalarObject

Reply:
See comment above

Reply:
Yup